crates/amaru-kernel/src/lib.rs

```diff
@@ -1019,6 +1088,21 @@ pub trait HasScriptHash {
     fn script_hash(&self) -> ScriptHash;
 }
 
+impl HasScriptHash for MintedScriptRef<'_> {
+    fn script_hash(&self) -> ScriptHash {
+        match self {
+            PseudoScript::NativeScript(native_script) => {
+                let mut buffer: Vec<u8> = vec![0];
+                buffer.extend_from_slice(native_script.raw_cbor());
+                Hasher::<224>::hash(&buffer)
+            }
+            PseudoScript::PlutusV1Script(plutus_script) => plutus_script.script_hash(),
+            PseudoScript::PlutusV2Script(plutus_script) => plutus_script.script_hash(),
+            PseudoScript::PlutusV3Script(plutus_script) => plutus_script.script_hash(),
```

Comment:
```suggestion
            PseudoScript::PlutusV1Script(plutus_script) |
            PseudoScript::PlutusV2Script(plutus_script) |
            PseudoScript::PlutusV3Script(plutus_script) => plutus_script.script_hash(),
```

I've heard that it's better to write like this 😉